doc/user-guides/secure-protect-connect-single-multi-cluster.md

```diff
@@ -0,0 +1,703 @@
+
+```bash
+cat $oasPath | envsubst | kuadrantctl generate gatewayapi httproute --oas -
+```
+
+If we're happy with the generated resource, let's apply it to the cluster:
+
+```bash
+cat $oasPath | envsubst | kuadrantctl generate gatewayapi httproute --oas - | kubectl apply -f -
+```
+
+Check our new route:
+
+```bash
+kubectl get httproute toystore -n ${$devNS} -o=yaml
+```
+
+We should see that this route is affected by the `AuthPolicy` and `RateLimitPolicy` defined as defaults on the gateway in the gateway namespace.
+
+```yaml
+- lastTransitionTime: "2024-04-26T13:37:43Z"
+        message: Object affected by AuthPolicy demo/external
+        observedGeneration: 2
+        reason: Accepted
+        status: "True"
+        type: kuadrant.io/AuthPolicyAffected
+- lastTransitionTime: "2024-04-26T14:07:28Z"
+        message: Object affected by RateLimitPolicy demo/external
+        observedGeneration: 1
+        reason: Accepted
+        status: "True"
+        type: kuadrant.io/RateLimitPolicyAffected        
+```
+
+### Test connectivity and deny-all auth 
+
+We'll use `curl` to hit an endpoint in the toystore app. As we are using LetsEncrypt staging in this example, we pass the `-k` flag:
+
+```bash
+curl -s -k -o /dev/null -w "%{http_code}" "https://$(kubectl get httproute toystore -n ${$devNS} -o=jsonpath='{.spec.hostnames[0]}')/v1/toys"
+```
+
+We are getting a `403` because of the existing default, deny-all `AuthPolicy` applied at the Gateway. Let's override this for our `HTTPRoute`.
+
+Choose one of the following options:
+
+### API key auth flow
+
+Set up an example API key in our cluster(s):
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: toystore-api-key
+  namespace: ${$devNS}
+  labels:
+    authorino.kuadrant.io/managed-by: authorino
+    kuadrant.io/apikeys-by: api_key
+stringData:
+  api_key: secret
+type: Opaque
+EOF
+```
+
+Next, generate an `AuthPolicy` that uses secrets in our cluster as APIKeys:
+
+```bash
+cat $oasPath | envsubst | kuadrantctl generate kuadrant authpolicy --oas -
+```
+
+From this, we can see an `AuthPolicy` generated based on our OAS that will look for API keys in secrets labeled `api_key` and look for that key in the header `api_key`. Let's now apply this to the gateway:
+
+```bash
+cat $oasPath | envsubst | kuadrantctl generate kuadrant authpolicy --oas -  | kubectl apply -f -
+```
+
+We should get a `200` from the `GET`, as it has no auth requirement:
+
+```bash
+curl -s -k -o /dev/null -w "%{http_code}" "https://$(kubectl get httproute toystore -n ${$devNS} -o=jsonpath='{.spec.hostnames[0]}')/v1/toys"
+```
+
+We should get a `401` for a `POST` request, as it does not have any auth requirements:
+
+```bash
+curl -XPOST -s -k -o /dev/null -w "%{http_code}" "https://$(kubectl get httproute toystore -n ${$devNS} -o=jsonpath='{.spec.hostnames[0]}')/v1/toys"
+```
+
+Finally, if we add our API key header, with a valid key, we should get a `200` response:
+
```

Comment:
```suggestion
From this, you can see an `AuthPolicy` generated based on your OAS that will look for API keys in secrets labeled `api_key` and look for that key in the header `api_key`. You can now apply this to the Gateway:

```bash
cat $oasPath | envsubst | kuadrantctl generate kuadrant authpolicy --oas -  | kubectl apply -f -
```

You should get a `200` from the `GET` because it has no auth requirement:

```bash
curl -s -k -o /dev/null -w "%{http_code}" "https://$(kubectl get httproute toystore -n ${$devNS} -o=jsonpath='{.spec.hostnames[0]}')/v1/toys"
```

You should get a `401` for a `POST` request because it does not have any auth requirements:

```bash
curl -XPOST -s -k -o /dev/null -w "%{http_code}" "https://$(kubectl get httproute toystore -n ${$devNS} -o=jsonpath='{.spec.hostnames[0]}')/v1/toys"
```

Finally, if you add your API key header, with a valid key, you should get a `200` response:

```